ledger/ledger-api-tests/suites/src/main/scala/com/daml/ledger/api/testtool/suites/v1_8/ActiveContractsServiceIT.scala

```diff
@@ -672,7 +678,17 @@ class ActiveContractsServiceIT extends LedgerTestSuite {
       c1 <- ledger.create(party, Dummy(party))
       anOffset <- ledger.currentEnd()
       _ <- ledger.create(party, Dummy(party))
-      _ <- ledger.prune(pruneUpTo = anOffset)
+      _ <- FutureAssertions.succeedsEventually(
+        retryDelay = 100.millis,
+        maxRetryDuration = 10.seconds,
+        ledger.delayMechanism,
+        "Pruning",
+      ) {
```

Comment:
Could you explain on the PR message why it is necessary to have retries on canton?

Reply:
And why do we need a double retry mechanism.  i.e. why not just keep the call we had above but add `attempts = 100` ?

Reply:
And the reason for the `attempts` is explained in the `SingleParticipantTestContext` method: "Distributed ledger participants need to reach global consensus prior to pruning"

Reply:
It has to do with acs commitments. My understanding is that pruning won't succeed if acs commitments haven't been exchanged. And submitting a command is encouraging Canton to exchange them.
That explains retries.

As to specific timeout values I repeated the approach from https://github.com/digital-asset/daml/blob/main/ledger/ledger-api-tests/suites/src/main/scala/com/daml/ledger/api/testtool/suites/v1_8/CommandDeduplicationPeriodValidationIT.scala#L182-L193

> And why do we need a double retry mechanism. i.e. why not just keep the call we had above but add attempts = 100 ?

Let me test it and see what happens

Reply:
> And why do we need a double retry mechanism. i.e. why not just keep the call we had above but add attempts = 100 ?

The inner retry mechanism is useless here. What we want to retry pruning + command submission which is impossible to do with the inner retries. Thus, we effectively disable the inner one by setting `attempts = 1`.

Reply:
As trying to encourage an ACS committment exchange is a bit subtle it may be worth a comment of factoring out that loop into a suitably named method.